### ISSUE TITLE
Mouse Lagging and Stuttering

### ISSUE THREAD

**Issue by MiniFullC** [carlrobertoh/CodeGPT#88]
After adding the markdown code block feature, I noticed that the mouse response when selecting text is very slow and feels laggy. 

Which IDEA diagnostic information should I provide to you for feedback?

os:  window11 21H2
version:  CodeGPT [1.10.3](https://plugins.jetbrains.com/plugin/21056-codegpt/versions/stable/323325)
![Animation-21 07345d](https://user-images.githubusercontent.com/37063727/233674934-75465b93-afbc-4b3c-9e84-36a722e09492.gif)

**Comment by carlrobertoh** [comment]
Thank you for reporting! 
Just to be clear, is it happening all the time or randomly?

**Comment by MiniFullC** [comment]
It seems that there is no lag when starting IDEA afresh, but it starts to lag after running the project.

**Comment by carlrobertoh** [comment]
I added an off-screen browser rendering support which you can toggle in from the settings (CodeGPT | Configuration).
Please let me know if it has any effect.

**Comment by MiniFullC** [comment]
![Animation-21 07345d](https://user-images.githubusercontent.com/37063727/236375630-888e59b0-923d-4d13-a69e-7ff460751b4d.gif)
Still the same problem, but with progress.
I'm opening multiple projects (IDEA windows) and it seems that the first project doesn't freeze, is this related?

**Comment by carlrobertoh** [comment]
Hey, I forgot to mention that after enabling the option, you still need to open either a new conversation or restart the IDE.

**Comment by MiniFullC** [comment]
Restarting the ide doesn't improve either.
It looks like it occurs several hours after running the IDE.

**Comment by carlrobertoh** [comment]
Yes, I also have experienced some lag after the IDE has been running for some time.
Not sure what is causing it yet.

**Comment by dragetd** [comment]
I was about to open another issue when I found this. Somehow I did not notice this the first time trying the extension (~3 weeks ago). Either because I remained in the chat window or it was only introduced somewhere inbetween.

But when working with the IDE while chat-completions are being created, in 90% of the time the whole IDE becomes unresponsive in certain ways. I.e. right-click for context-menu in my code view does not work at all. The 'copy' or 'replace with selection' button in CodeGPT also do not respond at first and only after several seconds register a click. Copying via 'ctrl-c' does not work from the CodeGPT window, tho this might be unrelated.

The moment I close the CodeGPT tab everything is normal again.

So yeah, it might be something about the browser component trashing the UI thread.

Only through this issue I found 'off-screen browser rendering support' which indeed improved the issue for me a lot. The IDE remains responsive. The CodeGPT window is still lagging a little (i.e. delay after clicking 'stop generating' or such), but much better to work with than before.

Any reason not to us off-screen-rendering (web-view in a different thread?) by default?

For reference:
* Gentoo Linux x64 on Kernel 6.1
* Intel GPU on MESA 22.3 / Wayland KDE
* IDEA 2023.1.1

**Comment by FichteFoll** [comment]
I have experienced similar issues with the codegpt panel in general becoming less responsive after a while, even when it was closed and reopened before, and also with all my keyboard inputs made into the main editor being swallowed until the panel is closed. This includes key bindings such as ctrl+v but also the insertion of normal characters.

**Comment by carlrobertoh** [comment]
I haven't figured out what exactly is causing it yet.

Off-screen rendering allows the browser to pre-load content in the background before it becomes visible to the user.

> Any reason not to us off-screen-rendering (web-view in a different thread?) by default?

The only reason I see so far is that you'll lose the popup menu functionality and the scrolling behaviour is bit different. For me, it makes the UX bit worse than the on-screen rendering.

Another thing you might want to try is to turn off GPU hardware acceleration by going to the IDE's Registry and searching for the flag `ide.browser.jcef.gpu.disabled`. Altho I'm not sure if it has any effect but worth to try.

Furthermore, could you profile the CPU usage and send me the logs?
(Help | Diagnostic Tools | Start CPU Usage Profiling)

@dragetd @FichteFoll

**Comment by carlrobertoh** [comment]
@MiniFullC @dragetd @FichteFoll 
Could you try upgrading the extension to the latest version and see if the problem still exists?
Thank you.

**Comment by dragetd** [comment]
Sorry, forgot about the profile dumps! Will test in two hours :)

**Comment by FichteFoll** [comment]
Haven't been able to reproduce it with 2.0.0 yet (which btw does not have a tag on this repo yet), thanks.

**Comment by carlrobertoh** [comment]
Thank you for testing! Closing this issue now.
